lib/services/local/generic/generic153.go

```diff
@@ -0,0 +1,135 @@
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package generic
+
+import (
+	"context"
+	"time"
+
+	"github.com/gravitational/trace"
+
+	machineidv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/machineid/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/teleport/lib/services"
+)
+
+// NewService153 will return a new generic service for a given RFD 153-style resource.
+func NewService153[T types.ResourceMetadata](
+	backend backend.Backend,
+	resourceKind string,
+	backendPrefix string,
+	marshalFunc MarshalFunc[T],
+	unmarshalFunc UnmarshalFunc[T]) (ServiceCommon[T], error) {
+
+	cfg := &ServiceConfig[resourceMetadataAdapter[T]]{
+		Backend:       backend,
+		ResourceKind:  resourceKind,
+		PageLimit:     0, // use default page limit
+		BackendPrefix: backendPrefix,
+		MarshalFunc: func(w resourceMetadataAdapter[T], option ...services.MarshalOption) ([]byte, error) {
+			return marshalFunc(w.resource)
+		},
+		UnmarshalFunc: func(bytes []byte, option ...services.MarshalOption) (resourceMetadataAdapter[T], error) {
+			r, err := unmarshalFunc(bytes, option...)
+			return newResourceMetadataAdapter(r), trace.Wrap(err)
+		},
+	}
+	svc, err := NewService[resourceMetadataAdapter[T]](cfg)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &service153[T]{svc: svc}, nil
+}
+
+var _ ServiceCommon[*machineidv1.Bot] = (*service153[*machineidv1.Bot])(nil)
+
+// service153 is a service adapter for RFD 153-style resources.
+type service153[T types.ResourceMetadata] struct {
+	svc *Service[resourceMetadataAdapter[T]]
```

Comment:
nit: spell it out:

```suggestion
	service *Service[resourceMetadataAdapter[T]]
```

Reply:
I thought the consensus was that vowels are too expensive to be used as field names, yet an important field like that perhaps deserves the expense 😂